Realm/RLMObservation.mm

```diff
@@ -553,3 +555,83 @@ void RLMDidChange(std::vector<realm::BindingContext::ObserverState> const& obser
+                ck = table->get_column_key(property.columnName.UTF8String);
+                table = table->get_link_target(ck);
+            } else if (property.type == RLMPropertyTypeLinkingObjects) {
+                // If the property type is a backlink, subsequent keys must be retrieved from the origin table
+                NSString *originTableName = [NSString stringWithFormat:@"class_%@", property.objectClassName];
+                StringData originTableNameStringData = RLMStringDataWithNSString(originTableName);
+                TableRef originTable = info->realm->_realm->read_group().get_table(originTableNameStringData);
+                
+                // The property from the origin table that links to the current property in the target table
+                NSString *originForwardLinkColumnName = property.linkOriginPropertyName;
+                StringData originForwardLinkColumnStringData = RLMStringDataWithNSString(originForwardLinkColumnName);
+                ColKey forwardLinkColumnKey = originTable->get_column_key(originForwardLinkColumnStringData);
+                
+                // Get the opposite (backlinked) column from the origin table
+                ck = originTable->get_opposite_column(forwardLinkColumnKey);
+                // Reassign table to the origin table so subsequent
+                // iterations can access column keys from origin table
+                table = originTable;
+            } else {
+                // This branch should never be reached. This case should have been caught by precondition above.
+                // How should an error be properly thrown?
+                RLMException(@"Precondition failure");
+            }
+
+            keyPairs.push_back(std::make_pair(tk, ck));
+            objectSchema = schema[property.objectClassName];
+        }
+
+        start = end + 1;
+    } while (end != NSNotFound);
+
+    TableKey tk = table->get_key();
+    ColKey ck;
+    // TODO: refactor
+    if (property.type == RLMPropertyTypeLinkingObjects) {
+        // If the property type is a backlink, subsequent keys must be retrieved from the origin table
+        NSString *originTableName = [NSString stringWithFormat:@"class_%@", property.objectClassName];
+        StringData originTableNameStringData = RLMStringDataWithNSString(originTableName);
+        TableRef originTable = info->realm->_realm->read_group().get_table(originTableNameStringData);
+
+        // The property from the origin table that links to the current property in the target table
+        NSString *originForwardLinkColumnName = property.linkOriginPropertyName;
+        StringData originForwardLinkColumnStringData = RLMStringDataWithNSString(originForwardLinkColumnName);
+        ColKey forwardLinkColumnKey = originTable->get_column_key(originForwardLinkColumnStringData);
+
+        // Get the opposite (backlinked) column from the origin table
+        ck = originTable->get_opposite_column(forwardLinkColumnKey);
```

Comment:
This block is very similar to the part inside the do-while and can probably be exported?
Or maybe included in the do-while be going one step further in the array.